packages/react-dom-bindings/src/server/ReactFizzConfigDOM.js

```diff
@@ -2759,9 +2760,23 @@ function pushStyle(
         rules: ([]: Array<Chunk | PrecomputedChunk>),
         hrefs: [stringToChunk(escapeTextForBrowser(href))],
         sheets: (new Map(): Map<string, StylesheetResource>),
+        nonce: nonce && stringToChunk(escapeTextForBrowser(nonce)),
       };
       renderState.styles.set(precedence, styleQueue);
     } else {
+      if (!('nonce' in styleQueue)) {
+        // `styleQueue` could have been created by `preinit` where `nonce` is not required
+        styleQueue.nonce = nonce && stringToChunk(escapeTextForBrowser(nonce));
+      }
```

Comment:
perhaps we should just consult the the one from the `renderState`? I'm not sure if it's a user requirement to pass `nonce` to `renderToPipeableStream` though and then maintain the consistency across the tree .

Reply:
we've thought about whether we should auto-nonce styles and scripts but have held off b/c you aren't necessarily in control of every tag you emit and so it's a little safer to still require the author pass these nonces around. However it's not really great protection b/c if you have a compromised package you are probably in trouble in many other ways too.

That said the nonce argument is implied to be for scripts and while rare it's technically possible for the nonce for styles to be different than the nonce for scripts so we'd have to expose another new option. Worth considering perhaps but I think making it just work with rendered props is the way to go

Reply:
So my question wasn't about auto-noncing anything but rather about checking the `renderState.nonce`'s value. This way we could cache the chunk on it directly and reuse it here if the per-style nonce would match it. But I'm not sure if it is a requirement to pass `nonce` to `renderToPipeableStream` (and that's how `renderState` learns about the nonce in the first place).

Reply:
you need to pass nonce to renderToPipeableStream if you are going to use CSP to restrict script execution by nonce. This is because React emits scripts to implement streaming rendering. But we don't typically that this option as an invitation to apply nonces to all possible nonceable things that a user might render like style tags and their own scripts. It's still on you to provide the nonce for your own scripts